src/core/model/StrokeStyle.cpp

```diff
@@ -1,91 +1,82 @@
 #include "StrokeStyle.h"
 
 #include <cstring>  // for memcmp, strcmp, strncmp
+#include <iomanip>  // for setprecision
+#include <iterator> // for ostream_iterator
+#include <map>      // for map
+#include <sstream>  // for istringstream
 #include <vector>   // for vector
 
-#include <glib.h>  // for g_ascii_strtod, g_free, g_strdup_printf
-
 #include "model/LineStyle.h"  // for LineStyle
 
 StrokeStyle::StrokeStyle() = default;
 
 StrokeStyle::~StrokeStyle() = default;
 
-const double dashLinePattern[] = {6, 3};
-const double dashDotLinePattern[] = {6, 3, 0.5, 3};
-const double dotLinePattern[] = {0.5, 3};
-
-#define PARSE_STYLE(name, def)                                \
-    if (strcmp(style, name) == 0) {                           \
-        LineStyle style;                                      \
-        style.setDashes(def, sizeof(def) / sizeof((def)[0])); \
-        return style;                                         \
-    }
+namespace {
 
-auto StrokeStyle::parseStyle(const char* style) -> LineStyle {
-    PARSE_STYLE("dash", dashLinePattern);
-    PARSE_STYLE("dashdot", dashDotLinePattern);
-    PARSE_STYLE("dot", dotLinePattern);
+constexpr auto CUST_KEY = "cust: ";
```

Comment:
```suggestion
constexpr auto CUSTOM_KEY = "cust: ";
```
This is more explicit